smartapps/juano2310/jawbone-up-connect.src/jawbone-up-connect.groovy

```diff
@@ -77,20 +76,21 @@ def callback() {
 
 def authPage() {
     log.debug "authPage"
-    def description = null          
+    def description = null
     if (state.JawboneAccessToken == null) {
 		if (!state.accessToken) {
 			log.debug "About to create access token"
 			createAccessToken()
 		}
         description = "Click to enter Jawbone Credentials"
-        def redirectUrl = oauthInitUrl()
-        // log.debug "RedirectURL = ${redirectUrl}"
-        return dynamicPage(name: "Credentials", title: "Jawbone UP", nextPage: null, uninstall: true, install:false) {
-               section { href url:redirectUrl, style:"embedded", required:true, title:"Jawbone UP", description:description }
+        def redirectUrl = buildRedirectUrl
+        log.debug "RedirectURL = ${redirectUrl}"
+        def donebutton= state.JawboneAccessToken != null
+        return dynamicPage(name: "Credentials", title: "Jawbone UP", nextPage: null, uninstall: true, install: donebutton) {
+               section { href url:redirectUrl, style:"embedded", required:true, title:"Jawbone UP", state: hast ,description:description }
         }
     } else {
-        description = "Jawbone Credentials Already Entered." 
+        description = "Jawbone Credentials Already Entered."
         return dynamicPage(name: "Credentials", title: "Jawbone UP", uninstall: true, install:true) {
                section { href url: buildRedirectUrl("receivedToken"), style:"embedded", state: "complete", title:"Jawbone UP", description:description }
```

Comment:
We don't redirect there, it is the Service Manager settings. That is what you see if you go back to the SmartApp. Usually you don't go there but you can use it to uninstall it.